### PR TITLE
[#174294463] Join communities

### PR DESCRIPTION
**Pull request by DmitryBarskov**
https://www.pivotaltracker.com/story/show/174294463

### CODE REVIEW

File: assets/javascripts/initializers/communitarian.js.es6 [hunk 1]
```diff
@@ -27,6 +28,19 @@ function initializeCommunitarian(api) {
   });
 
   api.modifyClass("controller:navigation/categories", isHomePageField);
+
+  registerUnbound('compare', function(v1, operator, v2) {
+    let operators = {
+      '==': (l, r) => l == r,
```
Comment (IlgizGilmanov) [inline assets/javascripts/initializers/communitarian.js.es6]:
```suggestion
      '===': (l, r) => l === r,
```

File: assets/javascripts/discourse/templates/components/leave-community-button.hbs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{{d-button
```
Comment (IlgizGilmanov) [inline assets/javascripts/discourse/templates/components/leave-community-button.hbs:1]: Is it possible to combine this two components (`join-community-button` & `leave-community-button`) into one? They are almost identical.
Then the contents of the `set-category-notification-level` component can be transferred there

File: assets/javascripts/initializers/communitarian.js.es6 [hunk 3]
```diff
@@ -4,10 +4,59 @@ import { setDefaultHomepage } from "discourse/lib/utilities";
...
+    limitedCategories(categories, length) {
+      this.set(
+        "model.categories.content",
+        categories.content.slice(0, length || 4)
+      );
+      return categories;
+    },
+  });
+
+  api.modifyClass("controller:navigation/categories", isHomePageField);
```
Comment (IlgizGilmanov) [inline assets/javascripts/initializers/communitarian.js.es6]:
```suggestion

```

File: assets/javascripts/initializers/communitarian.js.es6 [hunk 4]
```diff
@@ -4,10 +4,59 @@ import { setDefaultHomepage } from "discourse/lib/utilities";
...
+      '!=': (l, r) => l != r,
+      '>':  (l, r) => l >  r,
+      '>=': (l, r) => l >= r,
+      '<':  (l, r) => l <  r,
+      '<=': (l, r) => l <= r,
+    };
+    return operators[operator] && operators[operator](v1, v2);
+  });
+
+  api.modifyClass("controller:discovery/categories", {
```
Comment (IlgizGilmanov) [inline assets/javascripts/initializers/communitarian.js.es6]: All `controller:discovery/categories` modifications can be completely removed from here (both places)

File: assets/javascripts/initializers/communitarian.js.es6 [hunk 5]
```diff
@@ -4,10 +4,59 @@ import { setDefaultHomepage } from "discourse/lib/utilities";
...
+    limitedCategories(categories, length) {
+      this.set(
+        "model.categories.content",
+        categories.content.slice(0, length || 4)
+      );
+      return categories;
+    },
+  });
+
+  api.modifyClass("controller:navigation/categories", isHomePageField);
```
Comment (IlgizGilmanov) [inline assets/javascripts/initializers/communitarian.js.es6]:
```suggestion

```

File: assets/javascripts/discourse/templates/components/leave-community-button.hbs (hunk 2 above)
Comment (DmitryBarskov) [reply assets/javascripts/discourse/templates/components/leave-community-button.hbs:1]: I don't know 😔

**Review by Vadim-Kurnatovskiy — APPROVED** [review]

File: assets/javascripts/discourse/templates/components/leave-community-button.hbs (hunk 2 above)
Comment (IlgizGilmanov) [reply assets/javascripts/discourse/templates/components/leave-community-button.hbs:1]: I will try to implement it in next PR

**Review by ShaidullinArtur — APPROVED** [review]
looks ok for me. I think that these components will be changed in the future since we will update community header soon for resolutions and ongoing dialogs pages

**Review by IlgizGilmanov — APPROVED** [review]

**Pull request merged**